dra-evolution/testdata/classes.yaml

```diff
@@ -1,162 +1,208 @@
-# (e.g., bandwidth=1G) and then create an aggregate class from those. This is
-# not great, but maybe acceptable for now.
-#
-# *Claims* can still add the Constraints; we just can't do it in the class.
-#######
-#
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.quantityAttributes['bandwidth.nic.k8s.io'].compareTo(quantity('1G')) >= 0"
+---
+# Request any 1Gbps or faster SR-IOV NIC, regardless of the vendor.
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
+metadata:
+  name: example.com-sriov-nic-1g-or-faster
+request:
+  requirements:
+  - device:
+      selector: "'bandwidth.nic.k8s.io' in device.attributes && device.quantityAttributes['bandwidth.nic.k8s.io'].compareTo(quantity('1G')) >= 0"
+---
 # Request an example.com PF
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: example.com-sriov-pf
-spec:
-  driver: example.com-sriov
-  constraints: "device.sriovType == 'pf'"
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.stringAttributes['type.nic.k8s.io'] == 'pf'"
 ---
 # Request an example.com VF
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: example.com-sriov-vf
-spec:
-  driver: example.com-sriov
-  constraints: "device.sriovType == 'vf'"
----
-#######
-# NOTE: We cannot create a class for example.org PF or VF, because instead of
-# using attributes, they use custom vf/pf resource types to control this. Claims
-# can request one or the other by using `requests`.
-#######
-#
-# ConfigMap to configure a DRA driver that manages VLAN interfaces.
-# NOTE: How do we control permissions on what these configs can do?
-apiVersion: v1
-kind: ConfigMap
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.stringAttributes['type.nic.k8s.io'] == 'vf'"
+---
+# Request an example.com PF or VF
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
-  namespace: default
-  name: vlan-driver-vlan-2000
-data:
-  vlanID: 2000
+  name: example.com-sriov-vf
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.stringAttributes['type.nic.k8s.io'] == 'vf' || device.stringAttributes['type.nic.k8s.io'] == 'pf'"
 ---
 # Request a VLAN interface on VLAN 2000
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: vlan-2000
-spec:
-  deviceType: vlan
-  configs:
-  - apiVersion: v1
-    kind: ConfigMap
-    namespace: default
-    name: vlan-driver-vlan-2000
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.stringAttributes['type.k8s.io'] == 'vlan'"
+claim:
```

Comment:
Mentioned it earlier, but I find it super confusing to have the word "claim" in the class.